src/PowerShellEditorServices/Services/DebugAdapter/Handlers/DisconnectHandler.cs

```diff
@@ -48,10 +48,11 @@ internal class DisconnectHandler : IDisconnectHandler
         public async Task<DisconnectResponse> Handle(DisconnectArguments request, CancellationToken cancellationToken)
         {
             _debugEventHandlerService.UnregisterEventHandlers();
-            if (_debugStateService.ExecutionCompleted == false)
+
+            if (!_debugStateService.ExecutionCompleted)
             {
                 _debugStateService.ExecutionCompleted = true;
-                _executionService.CancelCurrentTask();
+                _debugService.Abort();
```

Comment:
There's a big question here still about ordering and what should happen.

Currently we call `UnregisterEventHandlers()` and then if execution is still going we call `Abort()` which stops a bunch of debug processes, or at least cancels out of a few of them.

But it's not clear we're handling *all* debug threads properly. We don't have cancellations set up for everything. It's also not clear whether unregistering things and *then* aborting debugging is the right order to do things in.

I'm reluctant to undertake more implementation here since it will only get more complex, but I think we should keep an eye on this. Basically my suspicion is that we might need to lock some things here, evaluate what's going on, and then ensure they've decided to cancel or not before releasing the lock.